### PR TITLE
build: create .gitignore

### PR DESCRIPTION
**Pull request by francisfuzz**
## Summary

Fixes https://github.com/copilot-extensions/preview-sdk.js/issues/69

This adds a `.gitignore` to the repository, sourced from https://github.com/github/gitignore/blob/main/Node.gitignore 😉

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+# Stores VSCode versions used for testing VSCode extensions
+.vscode-test
+
+# yarn v2
+.yarn/cache
+.yarn/unplugged
+.yarn/build-state.yml
+.yarn/install-state.gz
+.pnp.*
```
Comment (gr2m) [inline .gitignore:130]: I think there is so much stuff in here that we will never need. Let's just start with this

```suggestion
node_modules
```

And then we take it from there. It's a problem when it's a problem

**Review by gr2m — CHANGES_REQUESTED** [review]

**Comment by gr2m** [comment]
🤔 no idea why CI is not triggered for PRs coming from forks?

**Review by gr2m — APPROVED** [review]

**Comment by francisfuzz** [comment]
> 🤔 no idea why CI is not triggered for PRs coming from forks?

@gr2m - Great question! From [Controlling changes from forks to workflows in public repositories](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#controlling-changes-from-forks-to-workflows-in-public-repositories):

> Although workflows from forks do not have access to sensitive data such as secrets, they can be an annoyance for maintainers if they are modified for abusive purposes.
>
> To help prevent this, workflows on pull requests to public repositories from some outside contributors will not run automatically, and might need to be approved first. By default, all first-time contributors require approval to run workflows.

**Comment by gr2m** [comment]
> and might need to be approved first

Right, I remember the UI for this, I see it on other projects. But didn't see it for the PRs in this repository. I'm probably on some staff flag that turns on an alternative UI that does not have that feature implemented yet

**Comment by gr2m** [comment]
Confirmed, I see it in the classic view:
image

Good to know, will just switch back-and-forth as needed. Thank you @francisfuzz!!

**Pull request merged**